src/gui/widgets/group.hpp

```diff
@@ -0,0 +1,89 @@
+/*
+   Copyright (C) 2008 - 2016 The Battle for Wesnoth Project http://www.wesnoth.org/
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY.
+
+   See the COPYING file for more details.
+*/
+
+#ifndef GUI_WIDGETS_GROUP_HPP_INCLUDED
+#define GUI_WIDGETS_GROUP_HPP_INCLUDED
+
+#include "gui/auxiliary/event/dispatcher.hpp"
+#include "gui/widgets/selectable.hpp"
+#include "gui/widgets/widget.hpp"
+#include "utils/foreach.tpp"
+
+#include <vector>
+#include <boost/bind.hpp>
+
+namespace gui2
+{
+
+template <class T>
+class tgroup
+{
+public:
+	typedef typename std::vector<std::pair<tselectable_*, T> > group_list;
+	typedef typename group_list::iterator                      group_iterator;
+	typedef typename group_list::const_iterator                group_iterator_const;
+
+	// Setters for the member vector
+	void add_member(tselectable_* widget, const T& value)
+	{
+		members_.push_back(std::make_pair(widget, value));
+
+		dynamic_cast<twidget*>(widget)->connect_signal<event::LEFT_BUTTON_CLICK>(boost::bind(
+			&tgroup::group_operator, this), event::tdispatcher::front_child);
+	}
+
+	void remove_member(tselectable_* widget)
+	{
+		// TODO: implement
+		// members_.erase(std::find(members_.begin(), members_.end(), widget));
+	}
+
+	// Getters for the member vector
+	std::pair<group_iterator, group_iterator> members()
+	{
+		return std::make_pair(members_.begin(), members_.end());
+	}
+
+	std::pair<group_iterator_const, group_iterator_const> members() const
+	{
+		return std::make_pair(members_.begin(), members_.end());
+	}
+
+	void group_operator()
+	{
+		FOREACH(AUTO& member, members())
+		{
+			member.first->set_value(false);
+		}
+	}
+
+	T get_active_member_value()
```

Comment:
Hm... how would that work? Are you referring to setting the second member of the pair, or setting its t/f value?